sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2538,7 +2538,7 @@ object functions {
    * @since 1.5.0
    */
   def regexp_replace(e: Column, pattern: String, replacement: String): Column = withExpr {
-    RegExpReplace(e.expr, lit(pattern).expr, lit(replacement).expr)
+    new RegExpReplace(e.expr, lit(pattern).expr, lit(replacement).expr)
```

Comment:
Could you define a companion object to avoid this update?

Reply:
OK